umapi_client/functional.py

```diff
@@ -111,28 +82,27 @@ def __init__(self, id_type=IdentityTypes.adobeID, email=None, username=None, dom
             raise ArgumentError("Identity type (%s) must be one of %s" % (id_type, [i.name for i in IdentityTypes]))
         self.id_type = id_type
         self.email = None
-        self.domain = None
-        if username:
+        self.domain = domain
+        if username is not None:
             if email and username.lower() == email.lower():
                 # ignore the username if it's the same as the email (policy default)
                 username = None
-            elif id_type is not IdentityTypes.federatedID:
-                raise ArgumentError("Username must match email except for Federated ID")
-            else:
-                self._validate(username=username)
                 if domain:
```

Comment:
you've already unconditionally set `self.domain = domain` at line 85 so lines 90 and 91 do nothing.

Reply:
I'll initialize it to None to make it consistent with how it worked prior to this update.